pgscatalog.core/src/pgscatalog/core/lib/_normalise.py

```diff
@@ -121,101 +114,43 @@ def drop_hla(variants):
+    >>> list(assign_other_allele([variant]))[0] # doctest: +ELLIPSIS
+    ScoreVariant(..., effect_allele=Allele(allele='A', is_snp=True), other_allele=None, ...)
     """
     n_dropped = 0
     for variant in variants:
-        try:
-            if "/" in variant.other_allele:
-                n_dropped += 1
-                variant.other_allele = None
-        except TypeError:
-            pass  # other allele is already missing
-        finally:
-            yield variant
+        if getattr(variant.other_allele, "has_multiple_alleles", False):
+            n_dropped += 1
+            variant.other_allele = None
+        yield variant
 
     if n_dropped > 0:
         logger.warning(f"Multiple other_alleles detected in {n_dropped} variants")
         logger.warning("Other allele for these variants is set to missing")
 
 
-def assign_effect_type(variants):
-    """Convert PGS Catalog effect type columns to EffectType enums
-
-    The most common type of effect type is additive:
-
-    >>> variant = ScoreVariant(**{"effect_allele": "A", "effect_weight": 5, "accession": "test", "row_nr": 0, "is_recessive": "False", "is_dominant": "False"})
-    >>> list(assign_effect_type([variant])) # doctest: +ELLIPSIS
-    [ScoreVariant(...,effect_type=EffectType.ADDITIVE,...)]
-    >>> variant = ScoreVariant(**{"effect_allele": "A", "effect_weight": 5, "accession": "test", "row_nr": 0, "is_recessive": "True", "is_dominant": "False"})
-    >>> list(assign_effect_type([variant])) # doctest: +ELLIPSIS
-    [ScoreVariant(...,effect_type=EffectType.RECESSIVE,...)]
-    >>> variant = ScoreVariant(**{"effect_allele": "A", "effect_weight": 5, "accession": "test", "row_nr": 0, "is_recessive": "False", "is_dominant": "True"})
-    >>> list(assign_effect_type([variant])) # doctest: +ELLIPSIS
-    [ScoreVariant(...,effect_type=EffectType.DOMINANT,...)]
-
-    is_recessive and is_dominant fields are parsed from strings to bools during __init__.
-    """
-    for variant in variants:
-        match (variant.is_recessive, variant.is_dominant):
-            case (None, None) | (False, False) | (None, False) | (False, None):
-                # none is OK because is_recessive or is_dominant column may be missing
-                # default value is already set to additive, so just yield the variant
-                pass
-            case (False, True) | (None, True):
-                # none is OK because is_recessive column may be missing
-                variant.effect_type = EffectType.DOMINANT
-            case (True, False) | (True, None):
-                # none is OK because is_dominant column may be missing
-                variant.effect_type = EffectType.RECESSIVE
-            case _:
-                logger.critical(f"Bad effect type setting: {variant}")
-                raise Exception
-        yield variant
```

Comment:
This was deleted because effect_type is a computed and cached field in the pydantic model now